agent/consul/acl_endpoint_test.go

```diff
@@ -4342,9 +4342,17 @@ func TestACLEndpoint_SecureIntroEndpoints_OnlyCreateLocalData(t *testing.T) {
 		require.NoError(t, acl.Login(&req, &resp))
 		remoteToken = &resp
 
-		// present in dc2
-		resp2, err := retrieveTestToken(codec2, "root", "dc2", remoteToken.AccessorID)
-		require.NoError(t, err)
+		var resp2 *structs.ACLTokenResponse
+		var err error
+
+		// Flaky test! Need to retry/wait until !acl.srv.UseLegacyACLs() before continuing.
```

Comment:
Transitioning out of legacy ACL mode is a one-way-trigger. Waiting for both datacenters to transition (at the top of this test function) should be sufficient to guarantee we are out of legacy mode.

Reply:
done!